src/widgets.ts

```diff
@@ -28,12 +28,14 @@ export abstract class Widget<
   protected container: Element
   protected style: Partial<CSSStyleDeclaration>
   protected isUnmounting: boolean
+  protected ssoUrl: string
 
   // Filters for 'mx' events before dispatching to proper handlers
   protected messageCallback: (event: MessageEvent) => void
 
   constructor(options: WidgetOptions<Configuration, CallbackProps>) {
     this.isUnmounting = false
+    this.ssoUrl = ""
```

Comment:
Let's not initialize this to an empty string. Null value should represent a lack of a URL, and a string is a URL.